src/widgets/source/ui/CalibratorsPlotWidget.cpp

```diff
@@ -22,94 +22,376 @@
 // --------------------------------------------------------------------------
 
 #include <SmartPeak/ui/CalibratorsPlotWidget.h>
+#include <SmartPeak/core/SequenceSegmentProcessors/CalculateCalibration.h>
+#include <SmartPeak/core/Utilities.h>
 #include <implot.h>
 
 namespace SmartPeak
 {
+
+  void CalibratorsPlotWidget::recomputeCalibration()
```

Comment:
Overall, I think the updates to this widget are quite nice particularly the pop up of the chromatogram when selecting a point.  A couple of comments:

1. The SequenceSegment should be specified in e.g., a dropdown near the top of the widget where the component is specified.  This is because each sequentSegment can have its own quantitation method.

2. I would recommend two different user options for refitting: 1) OptimizeCalibration where all of the quantitation methods are recalculated using the SequenceSegment method `OptimizeCalibration`; and 2) ReCalculateCalibration where only the individual quantitation method is recalculated based on the previously optimized points and the points manually specified by the user to be included or excluded using the SequenceSegment method `CalculateCalibration`.

3. It might make sense to add a visual cue that alerts the user to the fact the the parameters are for the `OptimizeCalibration` method and the quantitation_method parameters are for both the `OptimizeCalibration` and `CalculateCalibration` methods.

Note that depending upon the user scenario, it might make sense to only allow for `CalculateCalibration`.  In that case, the parameters (i.e., for `OptimizeCalibration`) can be safely ignored and would not need to be shown to the user.

Reply:
so there is no possibility for the user to exclude points for the OptimizeCalibration?

Reply:
It should be possible for the user to exclude points for the OptimizeCalibration method.  Depending upon how the structure that links the standard with quantitation is built (see comment about making a separate method for that), an additional check to exclude user defined points should be possible.

This could cause problems down the road though if the user wanted to include all points for a fresh optimization.  So perhaps an option to add all points back in would be needed.

Reply:
Changed applied